components/n-ui/ads/js/utils.js

```diff
@@ -122,9 +122,25 @@ function getScreenSize () {
 	return window.innerWidth;
 }
 
-function inMetricsSample() {
-	return /spoor-id=3/.test(document.cookie);
-}
+// Fraction of all users that will actually send ads metrics to Spoor
+const METRICS_SAMPLE_SIZE = 0.1;
+
+const inMetricsSample = (() => {
+	let userSendsMetrics;
+	const decideInMetricsSample = () => {
+
+		// We are caching the value since we want to be consistent with the user
+		// allocation throughout one same session
+		if (typeof userSendsMetrics !== 'undefined') {
```

Comment:
Looks good, I'm just curious at how this userSendsMetrics is persisted, it's a bit difficult to track. but if I understand correctly, on subsequent calls you should still access the value if it has been defined for the current session?

Reply:
Yes. The innermost function (`defineInMetricsSample`) has a reference to the outer variable (`userSendsMetrics`) which creates a closure, (i.e. the variable is never garbage-collected) and its value is only accessible from within that function. Check out [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#Closure) for a better explanation.

Reply:
By the way, as Andrew pointed to me, 'session' is not correct here as the value is just persisted in browser memory for the current page visit. 

Navigating to a new page, will re-run the logic from scratch, so the user will be placed again in send/no-send group.